.circleci/config.yml

```diff
@@ -1,21 +1,73 @@
-version: 2
+version: 2.1
+
+orbs:
+  node: circleci/node@4.4.0
+
+workflows:
+  version: 2
+  workflow:
+    jobs:
+      - build:
+          matrix:
+            parameters:
+              # We should test against "latest Node 16", but there's an issue with 16.2
+              # This will be updated soon, along with a bug report to Node.js
+              # node-version: ["lts/erbium", "lts/fermium", "16"]
+              node-version: ["lts/erbium", "lts/fermium", "16.1"]
+      - release:
+          requires:
+            - build
+          filters:
+            branches:
+              only:
+                - main
+
 jobs:
   build:
-    docker:
-      - image: circleci/node:14.8.0
+    parameters:
+      node-version:
+        description: Node version to run tests under (set by matrix build config)
+        type: string
+    machine:
+      image: ubuntu-2004:202104-01
     steps:
       - checkout
-      - setup_remote_docker
-      - run: |
-          cd packages/integration-tests
-          docker-compose build db tests
-          docker-compose up db-wait
-          docker-compose up -d tests
-      - run: |
-          cd packages/integration-tests
-          docker-compose exec tests ./tests.sh
-      - run: |
-          cd packages/integration-tests
-          if [ "$CIRCLE_BRANCH" = "main" ]; then
-            docker-compose exec tests ./release.sh
-          fi
+      - node/install:
+          install-npm: false
+          node-version: << parameters.node-version >>
```

Comment:
The machine images have `nvm` pre-installed, so this step pulls the `nvm` node version from the matrix build and configures it as the default.